res/scripts/celmi/timetables/timetable.lua

```diff
@@ -1,809 +1,803 @@
-    local absDiff = math.abs(a - b)
-    if absDiff > 1800 then
-        return 3600 - absDiff
-    else
-        return absDiff
-    end
-end
-
----Shifts a time in minutes and seconds by some offset
----Helper function for shiftSlot() 
----@param time table in format like: {28,45}
----@param offset number in seconds 
----@return table shifted time, example: {30,0}
-function timetable.shiftTime(time, offset)
-    local timeSeconds = (time + offset) % 3600
-    return {math.floor(timeSeconds / 60), timeSeconds % 60}
-end
-
-
----Shifts a slot by some offset
----@param slot table in format like: {30,0,59,0}
----@param offset number in seconds 
----@return table slot shifted time, example: {31,0,0,0}
-function timetable.shiftSlot(slot, offset)
-    local arrivalSlot = timetable.slotToArrivalSlot(slot)
-    local shiftArr = timetable.shiftTime(arrivalSlot, offset)
-    local departureSlot = timetable.slotToDepartureSlot(slot)
-    local shiftDep = timetable.shiftTime(departureSlot, offset)
-    return {shiftArr[1], shiftArr[2], shiftDep[1], shiftDep[2]}
-end
-
--- removes old lines from timetable
-function timetable.cleanTimetable()
-    for lineID, _ in pairs(timetableObject) do
-        if not timetableHelper.lineExists(lineID) then
-            timetableObject[lineID] = nil
-            print("removed line " .. lineID)
-        else
-            local stations = timetableHelper.getAllStations(lineID)
-            for stationID = #stations + 1, #timetableObject[lineID].stations, 1 do
-                timetableObject[lineID].stations[stationID] = nil
-                print("removed station " .. stationID)
-            end
-        end
-    end
-end
-
-return timetable
-
+local timetableHelper = require "celmi/timetables/timetable_helper"
+local guard = require "celmi/timetables/guard"
+
+--[[
+timetable = {
+	line = {
+		stations = { stationinfo }
+		hasTimetable = true
+		frequency = 1 :: int
+	}
+}
+
+stationInfo = {
+	conditions = {condition :: Condition},
+	vehiclesWaiting = {
+		vehicleNumber = {
+			slot = {}
+			departureTime = 1 :: int
+		}
+	}
+}
+
+conditions = {
+	type = "None"| "ArrDep" | "debounce" | "moreFancey"
+	ArrDep = {}
+	debounce  = {}
+	moreFancey = {}
+}
+
+ArrDep = {
+	[1] = 12 -- arrival minute
+	[2] = 30 -- arrival second
+	[3] = 15 -- departure minute
+	[4] = 00 -- departure second
+}
+--]]
+
+local timetable = { }
+local timetableObject = { }
+
+function timetable.getTimetableObject()
+	return timetableObject
+end
+
+function timetable.setTimetableObject(t)
+	if t then
+		-- make sure the line is a number
+		local keysToPatch = { }
+		for lineID, lineInfo in pairs(t) do
+			if type(lineID) == "string" then
+				table.insert(keysToPatch, lineID)
+			end
+		end
+		
+		for _, lineID in pairs(keysToPatch) do
+			print("timetable: patching lineID: " .. lineID .. " to be a number")
+			local lineInfo = t[lineID]
+			t[lineID] = nil
+			t[tonumber(lineID)] = lineInfo
+		end
+		
+		timetableObject = nil

```

Comment:
For my understanding, can you clarify why setting this to nil is necessary?

Reply:
This is experimental. I wanted to remove any remnants of previous instances from the object. This way it is basically created from scratch.